### PR TITLE
km:0.1.0

### PR DESCRIPTION
**Pull request by E8D08F**
I am submitting
- [x] a new package
- [ ] an update for a package

Description: Draw diagrams to simplify boolean algebra expressions (implicants and don’t-cares are implemented).

I have read and followed the submission guidelines and, in particular, I
- [x] selected a name that isn't the most obvious or canonical name for what the package does
- [x] added a `typst.toml` file with all required keys
- [x] added a `README.md` with documentation for my package
- [x] have chosen a license and added a `LICENSE` file or linked one in my `README.md`
- [x] tested my package locally on my system and it worked
- [x] `exclude`d PDFs or README images, if any, but not the LICENSE

### CODE REVIEW

**Review by elegaanz — COMMENTED** [review]
LGTM, just a tip if you want to improve your README a bit.

File: packages/preview/karnaugh/0.1.0/README.md [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+    (1, 2, 2, 1),
+  ),
+  (
+    (0, 1, 0, 0),
+    (0, 1, 1, 1),
+  )
+)
+```
+
+Samples are available in `sample.pdf`.
```
[the remark this answers is not in the document]
Comment (E8D08F) [reply packages/preview/karnaugh/0.1.0/README.md]: Thanks for the advice 🙇‍♀️. Included the `sample.pdf` link that will become available if merged.

**Comment by elegaanz** [comment]
Thank you :)

**Comment by elegaanz** [comment]
Wait, now that I think about it, I'm not sure the package name fits the requirements: it should not be the most obvious/canonical name for what the package does, to not have an unfair advantage against other similar packages. Could you please try to find a name with a bit more "spice"? You can just add a prefix/suffix, make a pun, or add an adjective to the current name. 
After that, I will merge. Sorry that I didn't tell you before.

**Comment by E8D08F** [comment]
> Wait, now that I think about it, I'm not sure the package name fits the requirements: it should not be the most obvious/canonical name for what the package does, to not have an unfair advantage against other similar packages. Could you please try to find a name with a bit more "spice"? You can just add a prefix/suffix, make a pun, or add an adjective to the current name. After that, I will merge. Sorry that I didn't tell you before.

Thank you once again for your review and patience. I’d like to clarify the name requirement. It seems that `karnaugh` might be perceived as a family name which is an inexplicit reference to Maurice Karnaugh’s invention (just like Jacobi can refer to the mathematician Carl Gustav Jacob Jacobi, whereas ‘Jacobian’ or in Deutsch ‘Jacobische’ in its adjective form, is the exact signifier for [the determinant/matrix](https://en.wikipedia.org/wiki/Jacobian_matrix_and_determinant) of his. Similarly, Leonhard Euler’s contributions span various fields, so would the package name `euler` be acceptable?

After searching, I also found another package named [`k-mapper`](https://typst.app/universe/package/k-mapper). It directly relates to the established abbreviation of Karnaugh-Veitch maps, which might be factually advantageous. Another package, [`nassi`](https://en.wikipedia.org/wiki/Nassi–Shneiderman_diagram) is permitted, as it provides creation of Nassi–Shneiderman diagrams named after the author, Isaac Nassi.

Additionally, I’m curious if transcription to another language from canonical name is allowed. For example, _tiaoma_ means bar code in Mandarin, reflecting the core functionality of [that package](https://typst.app/universe/package/tiaoma). Similarly, _roremu_, a direct Japanese transcription from ”lorem“, [generates placeholder text](https://typst.app/universe/package/roremu).

Your insights on these matters would be greatly appreciated.

**Comment by elegaanz** [comment]
A package named `euler` would probably have to be renamed if it was too directly related to one's of Euler's contributions. Here the situation is even less ambiguous, as Karnaugh's most famous work is the Karnaugh maps, and we rarely (if ever) use his name for anything else, as far as I know.

`k-mapper` is not `karnaugh-map` or even `k-map`. It is close enough to be discoverable by people searching for these terms, but leaves spaces for alternative packages. As for the `nassi` package, it indeed kind of break this rule, and it is unfortunate that I didn't catch that during review, but it is not because some package inadvertently use a canonical name that any new package can ignore this rule.

And regarding transliterations, they are considered fine, as we assume that the Typst community is mostly English-speaking (or at least, it is the language that people use when using and contributing to Typst and its ecosystem).

I hope that these explanations help you to understand more clearly why I am asking you to rename the package.

**Comment by E8D08F** [comment]
Thank you for your clarification. I want to emphasise that I’ve got no intention of being disrespectful or disregarding the rules but just feel a bit confused and unfair if the same requirements have not been consistently enforced for past packages. A potentially unacceptable name like `karnaugh`, although originally used to honour the inventor, is not supposed to bypass the rules. However, people might have different interpretation for terms like “obivious”, as I mention in last reply, which is subjective to be defined, because their significance may vary among different groups; “canonical” means it’s unambiguously equivalent to the object which I didn’t see `karnaugh` fell into that category. Providing explicit guidelines could save us all time (we’ve seen several discussions like this in pull requests). It is also unrealistic to assume that coördinators can understand every nuance behind each name, and it can be exhausting to continuously guard against contributors who may intentionally break rules.

I understand that the goal of disallowing obvious or canonical names is to promote equality. But I question whether names are a decisive factor in creating monopoly. In current preview stage, there’s no  system like GitHub stars to measure the quality of packages; as far as I know, there haven’t been a way to provide namespaces other than `@local`/`@preview`, or to rename packages. Both methods can be helpful, which is why I wonder if the current naming requirement is a temporary policy, as it scarifies simplicity and tidiness from the users’ perspective (people sometimes must remember abbreviations from unfamiliar languages). It may also hinder collaboration (consensus may not be reached over some prefixes/suffixes from package names).

I apologise if I counldn’t make my idea clear. As for the package name, is the abbreviation, `km`, acceptable?

**Comment by elegaanz** [comment]
Yes these requirements are only in place because there is only one namespace that can be used at the moment. At some point, people should be free to name their package however they want just by having them in their own namespaces. Setting up explicit guidelines is hard because, there are cases that we can't think of that may arise, and people don't always read the provided guidelines anyway. In the end, it is better if a human does the verification, even if it is not error-proof.

And yes, `km` seems to be a better name.

**Comment by E8D08F** [comment]
Renamed to `km`. Thank you for your time.

**Comment by elegaanz** [comment]
Thank you!

**Pull request merged**